src/block_neural.rs

```diff
@@ -358,94 +358,92 @@ impl<L: OptimizerTrait + 'static> BlockTrait for BlockNeuronLayer<L> {
 
             block_helpers::forward_backward(further_blocks, fb, pb, update);
 
-            if update {
-                if self.neuron_type == NeuronType::WeightedSum {
-                    // first we need to initialize inputs to zero
-                    // TODO - what to think about this buffer
-                    let mut output_errors: [f32; MAX_NUM_INPUTS] =
-                        MaybeUninit::uninit().assume_init();
-                    output_errors
-                        .get_unchecked_mut(0..self.num_inputs)
-                        .fill(0.0);
-
-                    let (input_tape, output_tape) = block_helpers::get_input_output_borrows(
-                        &mut pb.tape,
-                        self.input_offset,
-                        self.num_inputs,
-                        self.output_offset,
-                        self.num_neurons,
-                    );
-
-                    for j in 0..self.num_neurons as usize {
-                        if self.dropout != 0.0
-                            && *self.rng_scratchpad.get_unchecked(j) < self.dropout_threshold
-                        {
-                            continue;
-                        }
+            if update && self.neuron_type == NeuronType::WeightedSum {
+                // first we need to initialize inputs to zero
+                // TODO - what to think about this buffer
```

Comment:
I think the buffer is great 👍 :) 

Might as well remove this comment as it doesn't provide any useful information.